lib/vrl/stdlib/src/is_array.rs

```diff
@@ -0,0 +1,84 @@
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct IsArray;
+
+impl Function for IsArray {
+    fn identifier(&self) -> &'static str {
+        "is_array"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            kind: kind::ANY,
+            required: true,
+        }]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "array",
+                source: r#"is_array([1, 2, 3])"#,
+                result: Ok("true"),
+            },
+            Example {
+                title: "boolean",
+                source: r#"is_array(true)"#,
+                result: Ok("false"),
+            },
+            Example {
+                title: "null",
+                source: r#"is_array(null)"#,
+                result: Ok("false"),
+            },
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let value = arguments.required("value");
+
+        Ok(Box::new(IsArrayFn { value }))
+    }
+}
+
+#[derive(Clone, Debug)]
+struct IsArrayFn {
+    value: Box<dyn Expression>,
+}
+
+impl Expression for IsArrayFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        let v = self.value.resolve(ctx)?;
+        match v {
+            Value::Array(_) => Ok(value!(true)),
+            _ => Ok(value!(false)),
+        }
```

Comment:
This code is fine, but could be simplified using [matches!](https://doc.rust-lang.org/std/macro.matches.html) to something like:

```rust
Ok(value!(matches!(v, Value::Array(_))))
```

Reply:
At that point, you might as well collapse it all together 😄 

`Value` has `is_xxx()` already implemented, so you can do:

```rust
self.value.resolve(ctx).map(Value::is_array)
```